source/slang/slang-ir-reachability.cpp

```diff
@@ -54,6 +55,45 @@ namespace Slang
         }
 
     }
+    
+    bool ReachabilityContext::isInstReachable(IRInst* inst1, IRInst* inst2)
+    {
+        // If inst1 and inst2 are in the same block,
+        // we test if inst2 appears after inst1.
+        if (inst1->getParent() == inst2->getParent())
+        {
+            for (auto inst = inst1->getNextInst(); inst; inst = inst->getNextInst())
+            {
+                if (inst == inst2)
+                    return true;
+            }
+        }
+
+        // Special cases
+
+        // Target switches; treat as reachable from its cases
+        if (auto tswitch = as<IRTargetSwitch>(inst2))
+        {
+            HashSet<IRInst*> blocks;
+            for (Slang::UInt i = 0; i < tswitch->getCaseCount(); i++)
+                blocks.add(tswitch->getCaseBlock(i));
+
+            IRInst* tmp = inst1->getParent();
+            while (tmp && tmp != tswitch->getParent())
+            {
+                if (blocks.contains(tmp))
+                    return true;
+
+                // Branching instructions
+                if (tmp->firstUse)
```

Comment:
I don't understand what this is tracking.

Reply:
In that case we should be extending isBlockReachable to properly handle target switch insts along with other control flow.

Reply:
I have simplified the logic for this part to rely on `isBlockReachable` more. We can't exactly extended `isBlockReachable` to handle switches and control flow because then it would relax the reachability to the point where something within the control flow's block is accessible after the control flow.